File: get_in_this_house.py
#!/usr/bin/env python3
"""
get_in_this_house.py

Generates a stylized ASCII-art meme framing a custom message in an "In this house, we…" format.
Checks the total character length (including frames and message) against a user-defined maximum.
Exits with an error if the rendered meme exceeds that limit.
"""

import argparse
import sys

def main():
    parser = argparse.ArgumentParser(
        description="Frame a custom message in a playful ASCII-art house meme."
    )
    parser.add_argument(
        "-m", "--message",
        default="MESSAGE",
        help="Custom message to display between the ASCII frames (default: 'MESSAGE')"
    )
    parser.add_argument(
        "-l", "--max-length",
        type=int,
        default=300,
        dest="max_length",
        help="Maximum allowed character count for the full meme (default: 300)"
    )
    args = parser.parse_args()

    top = """\
┏┓
┃┃╱╲in
┃╱╱╲╲this
╱╱╭╮╲╲house
▔▏┗┛▕▔ we
╱▔▔▔▔▔▔▔▔▔▔╲"""

    bottom = """\
╱╱┏┳┓╭╮┏┳┓ ╲╲
▔▏┗┻┛┃┃┗┻┛▕▔"""

    # Assemble the meme
    meme = f"{top}\n{args.message}\n{bottom}"

    # Count characters (includes newlines)
    length = len(meme)

    if length <= args.max_length:
        print(meme)
    else:
        print(f"Error: {length} -gt {args.max_length}", file=sys.stderr)
        sys.exit(1)

if __name__ == "__main__":
    main()

File: get_in_this_house.pl
#!/usr/bin/env perl
use strict;
use warnings;
use Getopt::Long;

# Default values
my $message   = 'Your message';
my $doormat   = '';
my $maxLength = 300;

# Parse command-line options
GetOptions(
    'message=s'   => \$message,
    'doormat=s'   => \$doormat,
    'maxLength=i' => \$maxLength,
) or die "Error in command line arguments\n";

# ASCII frames
my $top = <<'TOP';
┏┓
┃┃╱╲in
┃╱╱╲╲this
╱╱╭╮╲╲house
▔▏┗┛▕▔ we
╱▔▔▔▔▔▔▔▔▔▔╲
TOP

my $bottom = <<'BOTTOM';
╱╱┏┳┓╭╮┏┳┓ ╲╲
▔▏┗┻┛┃┃┗┻┛▕▔
BOTTOM

# Handle doormat centering
if ($doormat ne '') {
    my $houseWidth = 13;
    my $doormatWidth = length($doormat);
    my $halfMat = int($doormatWidth / 2);

    if ($doormatWidth < $houseWidth) {
        my $padding = int($houseWidth / 2) - $halfMat;
        $doormat = (' ' x $padding) . $doormat;
    }
}

# Construct meme
my $meme = "$top\n$message\n$bottom";
$meme .= "\n$doormat" if $doormat ne '';

# Validate length
if (length($meme) <= $maxLength) {
    print "$meme";
} else {
    die length($meme) . " -gt $maxLength\n";
}
